### DISCUSSION TITLE
Photo Sphere Viewer with Webcam Support

### DISCUSSION THREAD

**Discussion by UUoocl** [mistic100/Photo-Sphere-Viewer#1407] [Show and tell]
Thanks for Photo-Sphere-Viewer.  It's a great library! 

I modified the [Equirectangular Video example](https://photo-sphere-viewer.js.org/guide/adapters/equirectangular-video.html#example) to add access to live Webcam or Desktop Capture video.  

The Insta360 X4 recently added webcam support.  The Photo Sphere Viewer can convert the the Equirectangular live feed from the X4 perfectly. 

https://github.com/UUoocl/PSV_Live_Video

The next feature I plan to add is MediaPipe Pose detection to dynamically pan and tile the camera view.

**Comment by mistic100** [comment]
I could cleanily allow such usage by allowing the video source to be an existing video element instead of a URL string

**Reply by mistic100** [reply]
In the next version, you will be able to provide a MediaStream object as `source`

```js
const stream = await navigator.mediaDevices.getUserMedia({ video: true });

const viewer = new Viewer({
    container: 'photosphere',
    adapter: [EquirectangularVideoAdapter, {
        autoplay: true,
        muted: true,
    }],
    panorama: {
        source: stream,
    },
});
```
